### DISCUSSION TITLE
random sidecars with another metrics

### DISCUSSION THREAD

**Discussion by vasyakrg** [cloudnative-pg/cloudnative-pg#1830] [Q&A]
I have an important metric to check the occupied space inside the pvc (i used longhorn)
for this I use sidecar joins to each psql cluster replica in other psql operators (zalando as example)

essentially, I'm checking how much space is left.
what about using cnpg ?
i didn't see in crd records the ability to join another sidecars with commonplace pvc

my metrics: https://github.com/vasyakrg/df-exporter

**Comment by sxd** [comment]
Hi @vasyakrg 

You mean like providing a metric with the space left in the storage of the database? in that case it's a big complicated since you may have the space of pgdata and the space for wal storage, currently we provide a metric with the name `cnpg_pg_database_size_bytes`
If you want to get the space used inside the PVC I'd suggest to get that metric from the CSI driver, or if you have an a idea on how we can provide that metric internally, will be also great, but need to consider that any system command may not be available since it could be a distroless image, even for PostgreSQL that's possible and a user could be using one like that.

In the meantime, I'm turning this into a discussion since doesn't look like an issue yet, but we can open an issue starting on the discussion.

**Comment by vasyakrg** [comment]
That was actually my idea.
We need a sidecar that will go exactly to pgdata and perform `dh -h` there
and give metrics from it.

I have the [sidecar](https://github.com/vasyakrg/df-exporter) itself ready, but no crd capability to plug it into the main pod of all replicas.

I think the idea of sidecars would be useful not only for this task

**Reply by sigmonsays** [reply]
any tips on doing this?
